### PR TITLE
fix: warehouse form field duplicate

### LINKED ISSUE #228: Bigquery warehouse creation with gcs loading method shows duplicate form fields
## Description
When we add a new BigQuery warehouse  and select the Loading method as `GCS` and then toggle it a few times from `Standard` to `GCS`, we see multiple form fields for the same input like `GCS Secret ID`.

## Goals
- [ ] This should not happen

## Expected Outcome
The form should not have duplicate fields after toggling the loading method

## Acceptance Criteria
- [ ] Form should not have duplicate fields after toggling the loading method 

## Implementation Details
The main logic is in the file `src/helpers/ConnectorConfigInput.tsx`

## Mockups / Wireframes
This should not happen:
image

### Product Name
Dalgo

### Project Name
Dalgo

### Organization Name:
Project Tech4Dev

### Domain
30  Others

### Tech Skills Needed:
JavaScript, React

### Mentor(s)
@fatchat 

### Complexity
High

### Category
Bug

### Sub Category
Frontend

### PR DESCRIPTION
**Pull request by huzaifmalik786**
## Summary
This PR fixes #228 . The problem of field duplicate when toggling between the 2 options has been fixed. The problem was that the parent of the input fields was not set properly which was causing the issue. 

## Goals Achieved
- [x] Form should not have duplicate fields after toggling the loading method

## Solution Demo
[screen-capture.webm](https://github.com/DalgoT4D/webapp/assets/100015095/7d98d268-32f0-4c3e-b729-77ad5c4fc284)

### CODE REVIEW

**Comment by fatchat** [comment]
@Ishankoradia it looks like this works!

File: src/helpers/ConnectorConfigInput.tsx [hunk 1]
```diff
@@ -288,7 +288,7 @@ class ConnectorConfigInput {
           // Check if the child has another level or not
           if (ele.specs && ele.enum && ele.enum.length === 0) {
             ele.specs.forEach((childEle: ConnectorSpec) => {
-              filteredChildSpecs.push({ ...childEle, order: ele.order });
+              filteredChildSpecs.push({ ...childEle, parent: childEle?.parent && childEle?.parent?.length > 0 ? childEle.parent : ele.parent, order: ele.order });
             });
```
Comment (Ishankoradia) [inline src/helpers/ConnectorConfigInput.tsx:292]: Hey @huzaifmalik786 why is this needed here `&& childEle?.parent?.length` . Wont the condition before `&&` handle all cases of `null`, `undefined` and empty string. 

Rest looks good to me !!

Comment (huzaifmalik786) [reply src/helpers/ConnectorConfigInput.tsx:292]: @Ishankoradia , i saw that the child element has empty string as parent. So, it's there to handle the case of empty string i.e. "".

Comment (huzaifmalik786) [reply src/helpers/ConnectorConfigInput.tsx:292]: My bad. It is working. I'll change it.

**Comment by huzaifmalik786** [comment]
@Ishankoradia , I removed the unnecessary conditional check.

**Review by Ishankoradia — APPROVED** [review]
